### DISCUSSION TITLE
publicRuntimeConfig in getServerSideProps

### DISCUSSION THREAD

**Discussion by ethanryan** [vercel/next.js#25792] [Help]
We are in the process of updating our pages from using getInitialProps to getServerSideProps and getStaticProps.

On some pages, we are using publicRuntimeConfig for some third-party script tags.

This page in the documentation says:

> A page that relies on publicRuntimeConfig must use getInitialProps to opt-out of Automatic Static Optimization. Runtime configuration won't be available to any page (or component in a page) without getInitialProps.

via: https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration

We've updated the page from using getInitialProps to getServerSideProps and tested the page with a production build both locally and in a dev environment, and the page's third-party script tags worked as expected.

Is the warning in the documentation incorrect about publicRuntimeConfig not be available with getServerSideProps?

Should that warning instead say:

> A page that relies on publicRuntimeConfig must use getInitialProps OR getServerSideProps to opt-out of Automatic Static Optimization. Runtime configuration won't be available to any page (or component in a page) without getInitialProps OR getServerSideProps.

**Comment by justinthings** [comment]
Same thing is happening to me, seems like you can access publicRuntimeConfig in any context, static or not. Not sure what the documentation means. I've tried accessing `publicRuntimeConfig` in each of these scenarios using `getServerSideProps`, 'getStaticProps', and 'getInitialProps':

```
λ  (Server)  server-side renders at runtime (uses getInitialProps or getServerSideProps)
○  (Static)  automatically rendered as static HTML (uses no initial props)
●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)
```
doing this:

```
import getConfig from 'next/config'
const {
  publicRuntimeConfig: { aVariableFromRuntimeConfig },
} = getConfig()

//then rendering aVariableFromRuntimeConfig on the page
```

build runs fine, and the page render the variable's value on the page just fine in all scenarios.

Is the documentation incorrect? Do we need to worry about using `publicRuntimeConfig` in static contexts?

**Reply by bsplosion** [reply]
This is also something I'd like to understand.  Are we running a risk somehow by referencing `publicRuntimeConfig` in static contexts?  It works fine right now so I suspect the documentation's warning is inaccurate, but the lack of a reply here is concerning.
